decidim-core/lib/decidim/events/extended_event.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Events
+    # Extends the BaseEvent to add common features to most events so you don't
+    # need to write each time the same code.
+    #
+    # The only convention you need to keep in mind is that the event name will be
+    # used as the i18n scope to search for the keys.
+    class ExtendedEvent < BaseEvent
```

Comment:
What about `LocalizedEvent`, `I18nEvent` or `TranslatedEvent` for this? It's mostly about dealing with auto-generated `I18n` keys.

Reply:
Also, maybe this should be a `Mixin` instead?

Reply:
But the `EmailEvent` and `NofiticationEvent` also add I18n stuff.

I'm not sure about Extended but coupling it with I18n doesn't make a lot of sense to me. This event will probably be like the default event since we usually want to send an Email and a Web notification, not sure what's the best naming here.

Reply:
Maybe I can apply the conventions I created here to `EmailEvent` and `NotificationEvent` and just keep some methods as i18n helpers and add them to `EmailEvent` and `NotificationEvent` (and get rid of extended event)

Reply:
What about `SimpleEvent`? If you need more configuration, then it's not `simple` and you'll need to inherit from `BaseEvent`, and `SimpleEvent` simplifies the creation of events

🤷‍♂️

Reply:
Works for me

Reply:
I like what you said before: Extend `EmailEvent` and `NotificationEvent` instead. It's more composable and make sense.

I find `SimpleEvent` a bit meh.

Reply:
Actually, I'm not sure I can do that. `EmailEvent` and `NotificationEvent` have some default values for the i18n keys and it doesn't work well with the refactor.

Reply:
`StandardEvent`, `DecidimEvent` ..?

Reply:
`StandardEvent` sounds good to me

Reply:
You know what, it's ok for me either case. I think `Events` will need a bit of iterating anyway - it feels like the API that's surfacing isn't good enough (not because of your refactor though, if anything improved things).

Reply:
Done